torch/csrc/distributed/c10d/Work.cpp

```diff
@@ -45,17 +46,17 @@ OpType Work::retrieveOpType() const {
 Work::~Work() = default;
 
 bool Work::isCompleted() {
-  std::lock_guard<std::mutex> lock(mutex_);
+  C10D_LOCK_GUARD(lock, mutex_);
   return completed_;
 }
 
 bool Work::isSuccess() const {
-  std::lock_guard<std::mutex> lock(mutex_);
+  C10D_LOCK_GUARD(lock, mutex_);
```

Comment:
wonder if it makes sense to change the name `lock` to instead be more descriptive?
We can do that later too.

Reply:
Right now we don't print the lock name -- just the mutex

This variable name only is used when condition variables are used as well